### DISCUSSION TITLE
Error running notebook 4a

### DISCUSSION THREAD

**Discussion by akandrew332** [angelolab/toffy#249] [Q&A] [answered]
I'm trying to run Rosetta compensation on 1 run with just 3 FOVs for starters.  In section 2, getting an error when trying to process the images for Rosetta compensation using the 3 different multipliers.  Error is posted below:
![RosettaCompensationError](https://user-images.githubusercontent.com/19846608/189180764-62fd20e4-9cbe-425b-96c6-6b1d5f7faa74.JPG)

**Comment by ngreenwald** [comment, marked as answer]
Hmm, it should automatically be adding that channel to your csv. This was after resolving the issue with automatically formatting your panel, right?

**Reply by akandrew332** [reply]
Yes, after the formatting error was corrected.  Also, I can see chan_39 in the toffy formatted panel csv.
![channelnotfound](https://user-images.githubusercontent.com/19846608/189197570-44889737-6c95-4e5d-b849-ebe65be4aa0c.JPG)

**Reply by akandrew332** [reply]
Our panel included potassium, might this be interfering with the downstream processing since chan_39 would also be potassium?

**Reply by camisowers** [reply]
Yes, that is what's happening. Since you have Potassium as a target for mass 39, the name was changed and the function cannot find a 'chan_39' target. A quick fix would be changing the channel name in the panel.

**Reply by ngreenwald** [reply]
Ah, that makes sense. Yeah, we previously never had anyone extract mass 39, so the script assumes it still has the default name. We'll update it to allow a user-specified name, but for now changing the name back to chan_39 should resolve the issue

**Reply by akandrew332** [reply]
Okay, so changing potassium to chan_39 in the panel.csv, and rerunning notebook 4a didn't work.  What did work was deleting all the FOV images from the extracted images folder on (D:) and then using the updated panel to rerun notebook 3b, and then rerunning 4a.  Then there is an extracted channel image for 4a to work with.  Notebook 4a actually works even if the panel.csv contains potassium vs chan_39, as long as the extracted images (on D:) have the expected names.  Additionally, if there are channels that cannot be compensated for in the original panel (like iodine, cadmium, etc..), you'll have to remove those (and the respective channels in the extracted images folder if 3b was run previously with the incorrect panel.csv) or you'll get the following error:
![RosettaCompensationerror2](https://user-images.githubusercontent.com/19846608/189235932-6a0408aa-83f7-4eb4-913b-6e9989fa0003.JPG)

**Reply by ngreenwald** [reply]
What are you guys interested in looking at with the naturally abundant masses? Is this data that you plan on analyzing? If so, you'll probably want gold correction etc. We can adjust the compensation script to skip over masses that aren't in the matrix, but if you're planning on analyzing them you'll probably want to go in and pick coefficients

**Reply by akandrew332** [reply]
We are ditching those channels at the moment now that we are starting to use Rosetta, so no worries.
